### DISCUSSION TITLE
One question about scoreDirector.getWorkingSolution method

### DISCUSSION THREAD

**Discussion by EasonCon** [TimefoldAI/timefold-solver#1098] [General]
Does Timefold make a copy of the solution and then perform neighborhood search on the copy? Is the solution object returned by scoreDirector.getWorkingSolution() a copy or the original solution?

**Comment by triceo** [comment]
Hello @EasonCon! 
Short answer: yes. 

Timefold Solver has its own clone on which it runs the search. If you're experiencing issues with cloning, perhaps you might want to [read the cloning documentation](https://docs.timefold.ai/timefold-solver/latest/using-timefold-solver/modeling-planning-problems#cloningASolution). You should only use the working solution to query its contents, never modify it.
